packages/sanity/src/ssr-test/hydrate-tests.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import ReactDOM from 'react-dom';
+import Registry from 'metadata-tools';
+import {hydrate} from 'react-dom';
+import chai, {expect} from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+
+chai.use(sinonChai);
+
+export const hydrateTests = (): void => {
+  describe('Hydration tests', () => {
+    let consoleSpy: sinon.SinonSpy;
+    let errorSpy: sinon.SinonSpy;
+    let index = 0;
+    const componentStrings = (window as any).components;
+    const root = document.getElementById('root');
+
+    Registry.metadata.components.forEach((metadata, Comp) => {
+      describe(Comp.name, () => {
+        metadata.simulations.forEach(((simulation) => {
+          beforeEach(() => {
```

Comment:
Don't really see a reason why this entire block shouldn't be in the test. The control flow would be more linear.